src/PermissionRegistrar.php

```diff
@@ -17,39 +17,27 @@ class PermissionRegistrar
     /** @var \Illuminate\Contracts\Cache\Repository */
     protected $cache;
 
-    /** @var \Illuminate\Contracts\Logging\Log */
-    protected $logger;
-
     /** @var string */
     protected $cacheKey = 'spatie.permission.cache';
 
-    public function __construct(Gate $gate, Repository $cache, Log $logger)
+    public function __construct(Gate $gate, Repository $cache)
     {
         $this->gate = $gate;
         $this->cache = $cache;
-        $this->logger = $logger;
     }
 
     public function registerPermissions(): bool
     {
-        try {
-            $this->getPermissions()->map(function (Permission $permission) {
-                $this->gate->define($permission->name, function ($user) use ($permission) {
-                    return $user->hasPermissionTo($permission);
-                });
-            });
-
-            return true;
-        } catch (Exception $exception) {
-            if ($this->shouldLogException()) {
-                $this->logger->alert(
-                    "Could not register permissions because {$exception->getMessage()}".PHP_EOL.
-                    $exception->getTraceAsString()
-                );
+        $this->gate->before(function (Authenticatable $user, string $ability) {
```

Comment:
Really great idea to leverage the gate's `before` function here. ❤️

Reply:
All credit goes to @hackel!

Reply:
I haven‘t it tested yet, but if you use the before method, you can‘t use it for example in the AuthServiceProvider like demonstrated here #384 for a super user. The before method in this example doesn‘t get called.

Reply:
@maxxscho The before method isn't a single-use hook, it adds the closure to a list of closures to run before the gate is checked.  If you haven't tested it yet, you should.  I can't think of any reason  other before calls shouldn't also work.

Reply:
@hackel I've tested it yesterday. I had a `Gate::before()` method in AuthServiceProvider to give all Permissions to a "SuperUser", similar to the example on #384 . The `before` method didn't get invoked.
Setup of Laravel Permission was with Standard-User class of Laravel and anything Auth-related was as described as in the Readme. I was looking for a mistake on my App-Code, but everything looks well und works perfectly, except the `Gate::before()`-method.

Reply:
This seems indeed has issue.via  this laravel gate `before()` callback invoke code
https://github.com/laravel/framework/blob/ac7eee2a13beb53b4806d1fb830f49dbf40df30c/src/Illuminate/Auth/Access/Gate.php#L290
it means although  the  added closure is a list of closures.but onces ths closure return not null value ,just like return true or false,it will immediately return the result and ignore the other closures.

and I looked the order regist of `Gate::before()`.this  spatie/laravel-permission alwasy before my 
`App\Providers\AuthServiceProvider`

Reply:
@stevenyangecho thanks for the hint. I took a look at the Gate class, but it seems not close enough 😄